src/worker-utils.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Fetch the given script and create a local URL for it
+ *
+ * @param {string} scriptUrl Real world location of the script
+ * @returns {Promise<string>} Local url for the script
+ */
+export async function getBlobUrl(scriptUrl) {
+  const text = scriptUrl.startsWith('ipfs')
+    ? await fetchIpfsUrl(scriptUrl)
+    : await fetchUrl(scriptUrl)
+  const blob = new Blob([text], { type: 'application/javascript' })
+
+  return URL.createObjectURL(blob)
```

Comment:
I don't expect this process (fetch script, stream into text, and blob it) to be too heavy performance-wise, but if we find people loading ridiculously big scripts (e.g. MBs), we can think about putting this functionality into a WebWorker that sends back either a blob url or transfers the whole blob back.